### PR TITLE
feat: openai encoder org id support

### PR DESCRIPTION
**Pull request by maxyousif15**
Adding support for `OPENAI_ORGANIZATION` when initialising the `OpenAIEncoder`.
This PR includes:
- The code changes to support this feature
- Minor updates to the unit tests to ensure this functionality works as expected

### CODE REVIEW

File: semantic_router/encoders/openai.py [hunk 1]
```diff
@@ -20,17 +20,19 @@ def __init__(
...
+        openai_org_id: Optional[str] = None,
         score_threshold: float = 0.82,
         dimensions: Union[int, NotGiven] = NotGiven(),
     ):
         if name is None:
             name = os.getenv("OPENAI_MODEL_NAME", "text-embedding-ada-002")
         super().__init__(name=name, score_threshold=score_threshold)
         api_key = openai_api_key or os.getenv("OPENAI_API_KEY")
-        if api_key is None:
+        org_id = openai_org_id or os.getenv("OPENAI_ORGANIZATION")
```
Comment (Siraj-Aizlewood) [inline semantic_router/encoders/openai.py:31]: I'm not sure we need a new `org_id` var, I think 

`openai_org_id = openai_org_id or os.getenv("OPENAI_ORGANIZATION")`

Should be okay.

File: semantic_router/encoders/openai.py [hunk 2]
```diff
@@ -20,17 +20,19 @@ def __init__(
...
+        if (api_key is None) and (org_id is None):
```
Comment (Siraj-Aizlewood) [inline semantic_router/encoders/openai.py:32]: We probably want to remove `and (org_id is None)`. If `api_key is None` the API call should not run, regardless of the value of `org_id`. I think it's also okay to make the API call if `org_id` is not set (I'm fairly certain it's optional), and so I wouldn't use 'or' here either. I think `and (org_id is None)` can just be remoed.

File: semantic_router/encoders/openai.py (hunk 1 above)
Comment (Siraj-Aizlewood) [inline semantic_router/encoders/openai.py:31]: For consistency, I'd rename `"OPENAI_ORGANIZATION"` to `"OPENAI_ORG_ID"`, as the latter is what's used by OpenAI in their code.

**Review by Siraj-Aizlewood — CHANGES_REQUESTED** [review]
Thanks for this @maxyousif15, just a few comments.

File: semantic_router/encoders/openai.py (hunk 1 above)
Comment (maxyousif15) [reply semantic_router/encoders/openai.py:31]: Great catch. I'm not sure where `OPENAI_ORGANIZATION` came from now, but it is all over my codebase so I will need to promptly update it.

**Review by Siraj-Aizlewood — COMMENTED** [review]
Previously reviewed. I'm happy to merge, but will check with James first.

**Review by jamescalam — APPROVED** [review]

**Review by Siraj-Aizlewood — APPROVED** [review]

**Comment by maxyousif15** [comment]
Any time. Thanks for this project. Looking forward to making more contributions down the line.

**Pull request merged**